script/deploy/query-deployment-logs.ts

```diff
@@ -0,0 +1,638 @@
+      },
+    }
+  }
+
+  public async findByAddress(
+    address: string
+  ): Promise<IDeploymentRecord | null> {
+    return this.collection.findOne({ address })
+  }
+
+  public async filterDeployments(filters: {
+    contractName?: string
+    network?: string
+    version?: string
+    verified?: boolean
+    limit?: number
+  }): Promise<IDeploymentRecord[]> {
+    const query: Record<string, unknown> = {}
+
+    if (filters.contractName) query.contractName = filters.contractName
+    if (filters.network) query.network = filters.network
+    if (filters.version) query.version = filters.version
+    if (filters.verified !== undefined) query.verified = filters.verified
+
+    return this.collection
+      .find(query)
+      .sort({ timestamp: -1 })
+      .limit(filters.limit || 50)
+      .toArray()
+  }
+
+  public async getDeploymentHistory(
+    contractName: string,
+    network: string
+  ): Promise<IDeploymentRecord[]> {
+    return this.collection
+      .find({ contractName, network })
+      .sort({ timestamp: -1 })
+      .toArray()
+  }
+
+  public async disconnect(): Promise<void> {
+    await this.client.close()
+  }
+}
+
+// Helper function to format deployment record for display
+function formatDeployment(deployment: IDeploymentRecord): string {
+  return `
+Contract: ${deployment.contractName}
+Network: ${deployment.network}
+Version: ${deployment.version}
+Address: ${deployment.address}
+Timestamp: ${deployment.timestamp.toISOString()}
+Verified: ${deployment.verified}
+Optimizer Runs: ${deployment.optimizerRuns}
+${deployment.salt ? `Salt: ${deployment.salt}` : ''}
+---`
+}
+
+// Helper function to output JSON without logging
+function outputJSON(data: unknown): void {
+  console.log(JSON.stringify(data, null, 2))
+}
+
+// Define latest command
+const latestCommand = defineCommand({
+  meta: {
+    name: 'latest',
+    description: 'Get latest deployment for a contract on a network',
+  },
+  args: {
+    env: {
+      type: 'string',
+      description: 'Environment (staging or production)',
+      default: 'production',
+    },
+    contract: {
+      type: 'string',
+      description: 'Contract name',
+      required: true,
+    },
+    network: {
+      type: 'string',
+      description: 'Network name',
+      required: true,
+    },
+  },
+  async run({ args }) {
+    // Validate environment
+    if (args.env !== 'staging' && args.env !== 'production') {
+      consola.error('Environment must be either "staging" or "production"')
+      process.exit(1)
+    }
+
+    const querier = new DeploymentLogQuerier(
+      config,
+      args.env as 'staging' | 'production'
+    )
+
+    try {
+      await querier.connect()
+      const deployment = await querier.getLatestDeployment(
+        args.contract,
+        args.network
+      )
+
+      if (deployment) outputJSON(deployment)
+      else process.exit(1)
+    } catch (error) {
+      consola.error('Query failed:', error)
+      process.exit(1)
```

Comment:
i think if the program stops right away and does not run any finally blocks, so await querier.disconnect() is skipped and the database connection stays open. I would call  await querier.disconnect() also before process.exit(1). Same for other cases with process.exit(). wdyt?

Reply:
fixed: 3298710023497f4a30595870fb38125c2f17787d